module/VuFind/src/VuFind/Db/Table/OaiResumption.php

```diff
@@ -85,11 +86,51 @@ public function removeExpired()
      *
      * @param string $token The resumption token to retrieve.
      *
-     * @return ?\VuFind\Db\Row\OaiResumption
+     * @return     ?\VuFind\Db\Row\OaiResumption
+     * @deprecated Use OaiResumption::findWithId
      */
     public function findToken($token)
     {
-        return $this->select(['id' => $token])->current();
+        return $this->findWithId($token);
+    }
+
+    /**
+     * Retrieve a row from the database based on primary key; return null if it
+     * is not found.
+     *
+     * @param string $id Id used for the search.
+     *
+     * @return ?\VuFind\Db\Row\OaiResumption
+     */
+    public function findWithId(string $id): ?OaiResumptionEntityInterface
+    {
+        return $this->select(['id' => $id])->current();
+    }
+
+    /**
+     * Retrieve a row from the database based on primary key and where the token is null.
+     *
+     * @param string $id Id used for the search.
+     *
+     * @return ?\VuFind\Db\Row\OaiResumption
+     * @todo   In future, we should migrate data to prevent null token fields, which will make this method obsolete.
+     */
+    final public function findWithTokenOrLegacyIdToken(string $id): ?OaiResumptionEntityInterface
```

Comment:
I think this should be `findWithLegacyIdToken` since this ONLY looks up by the legacy ID token, not by the regular token.

Reply:
Oh, my bad!